pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1193,11 +1193,6 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podIn
 
 	activated := false
 	for _, pInfo := range podInfoList {
-		// Since there may be many gated pods and they will not move from the
-		// unschedulable pool, we skip calling the expensive isPodWorthRequeueing.
-		if pInfo.Gated {
```

Comment:
/cc @alculquicondor @gabesaba 

Thanks for a detailed investigation in https://github.com/kubernetes/kubernetes/issues/125223#issuecomment-2170470530. This path is definitely problematic for PreEnqueue plugins other than schedulinggates, which PreEnqueue() could change its result by a change in resources apart from Pods. 😞

---

If we just revert it, we'd bring back the degradation raised at https://github.com/kubernetes/kubernetes/issues/124384. Also, the PR was cherry-picked to patch releases, which means we should make this PR simple enough to cherry-pick it too.

As a quick fix, can we use `pInfo.Gated && pInfo.UnschedulablePlugins.Has(names.SchedulingGates)` instead of problematic `pInfo.Gated`?

Reply:
Opened #125527, as a solution alternative to this PR.